src/corporacreator/preprocessors/cy.py

```diff
@@ -9,4 +9,23 @@ def cy(client_id, sentence):
       (str): Cleaned up sentence. Returning None or a `str` of whitespace flags the sentence as invalid.
     """
     # TODO: Clean up cy data
+
+    # make sure all spaces are normal, single spaces
+    sentence = sentence.replace("’","'")
+    # this if loop will skip the following else loop in the case that
+    # the sentence is purely Welsh characters as per https://github.com/
+    # unicode-cldr/cldr-misc-full/blob/master/main/cy/characters.json
+    if set(list(sentence)).issubset(set([' ', 'a','á', 'à', 'â', 'ä', 'b', 'c',
+                                         'd', 'e', 'é', 'è', 'ê', 'ë', 'f', 'g',
+                                         'h', 'i', 'í', 'ì', 'î', 'ï', 'j', 'l',
+                                         'm', 'n', 'o', 'ó', 'ò', 'ô', 'ö', 'p',
+                                         'r', 's', 't', 'u', 'ú', 'ù', 'û', 'ü',
+                                         'w', 'ẃ', 'ẁ', 'ŵ', 'ẅ', 'y', 'ý', 'ỳ',
+                                         'ŷ', 'ÿ'])):
+        pass
+    else:
+        # Not all chars are in the Welsh alphabet!
+        # more processing needed here!
+        pass
+
```

Comment:
There are several issues:

- Could you do something in the "else" case?
- Also, `sentence` is cased, but your alphabet is not.
- Also you re-create the set `set(['' ', 'a','á', 'à', 'â',...])` for each sentence. You should create is once and reuse it
- You don't need to do `set(list(sentence))` you only need `set(sentence)`
- This should be moved in to a function and not be inline in the `cy` function

All of this is also true of all the other PR's too.

Reply:
@kdavis-mozilla all your concerns have been addressed in newest commits to this branch. 

The only thing that I did different was using not a function to compile the `set()` of the alphabet, but just a single, first line in the file. I'm not sure of a better way to do this with a function.

Reply:
resolved in latest commit. No more alphabet-checking.